platforms/cuda/src/CudaTorchKernels.cpp

```diff
@@ -60,6 +66,13 @@ void CudaCalcTorchForceKernel::initialize(const System& system, const TorchForce
         globalNames.push_back(force.getGlobalParameterName(i));
     int numParticles = system.getNumParticles();
 
+    // Switch to the PyToch context
+    CUcontext context;
+    CHECK_RESULT(cuCtxGetCurrent(&context), "Failed to get the CUDA context");
+    if (context)
+        CHECK_RESULT(cuCtxSynchronize(), "Failed to synchronize the CUDA context"); // Synchronize before switching to the PyTorch context
+    CHECK_RESULT(cuCtxPushCurrent(primaryContext), "Failed to push the CUDA context");
+
```

Comment:
Can you explain what this code is for?  This method doesn't involve any communication between the PyTorch and OpenMM contexts, so synchronization shouldn't be necessary.  And as noted above, the primary context might not have even been created yet at this point.

Reply:
This ensures that `initialize` from a previous forces hasn't left some kernel running the OpenMM context. As, we seen before, bad thing happen if the context switches aren't synchronized. And as noted above, the primary context is activated in the constructor.

Reply:
There is no possibility of that happening.  This method is called immediately after the CudaContext is created, before `initialize()` has even been called yet on it.  We have a new, just created CUcontext.  Even if there were somehow a kernel running, it still wouldn't matter.  Nothing in this method does any communication between the PyTorch and OpenMM contexts.

Reply:
Correct.

Reply:
Removed that check (https://github.com/openmm/openmm-torch/pull/80/commits/d3d060b36f6b1bad07154f626f1fefe0e255fc0b)